src/compile/repeater.ts

```diff
@@ -44,16 +73,26 @@ function replaceRepeater(mapping: EncodingOrFacet<Field>, repeater: RepeaterValu
   const out: EncodingOrFacet<string> = {};
   for (const channel in mapping) {
     if (mapping.hasOwnProperty(channel)) {
-      const fieldDef: FieldDef<Field> | FieldDef<Field>[] = mapping[channel];
+      const channelDef: ChannelDef<Field> = mapping[channel];
 
-      if (isArray(fieldDef)) {
-        out[channel] = fieldDef.map(fd => replaceRepeaterInFieldDef(fd, repeater))
-          .filter((fd: FieldDef<string> | null) => fd !== null);
-      } else {
-        const fd = replaceRepeaterInFieldDef(fieldDef, repeater);
+      if (isArray<FieldDef<Field>>(channelDef)) {
+        out[channel] = channelDef.map(fd => replaceRepeaterInFieldDef(fd, repeater))
+          .filter(fd => fd !== null);
+      } else if (isConditionalDef(channelDef) && isFieldDef(channelDef.condition)) {
+        const fd = replaceRepeaterInFieldDef(channelDef.condition, repeater);
+        if (fd !== null) {
+          out[channel] = {
+            ...channelDef,
+            condition: fd
+          };
+        }
```

Comment:
In the else case, don't we have to at least do
```
out[channel] = channelDef;
```
? 

(Drop only the condition, not the whole thing)

Reply:
We could, I guess. Since there is a value, right?

Reply:
Yes, I think we definite should.

Reply:
Is this case handled somehow? (Doesn't look like it?)

Reply:
Done. There was also a missing case for reversed conditionals.

Reply:
Done